### ISSUE TITLE
Bot fails to auto-recover after a Discord server outage

### ISSUE THREAD

**Issue by norrisng** [norrisng/FcomServer#15]
After [the Discord outage on 2019-01-12 at 1050z](https://status.discordapp.com/incidents/cg7y5gbl80pg), the bot failed to forward any messages, even after Discord was back up and running.

In this case, the server threw a HTTP 500 error:

```
Task exception was never retrieved
future: <Task finished coro=<forward_messages() done, defined at .../FcomServer/bot/discord_bot.py:17> exception=HTTPException('INTERNAL SERVER ERROR (status code: 500): 500: Internal Server Error',)>
Traceback (most recent call last):
  File ".../FcomServer/bot/discord_bot.py", line 44, in forward_messages
    await dm_channel.send(dm_contents)
  File ".../FcomServer/venv/lib/python3.6/site-packages/discord/abc.py", line 764, in send
    data = yield from state.http.send_message(channel.id, content, tts=tts, embed=embed, nonce=nonce)
  File ".../FcomServer/venv/lib/python3.6/site-packages/discord/http.py", line 221, in request
    raise HTTPException(r, data)
discord.errors.HTTPException: INTERNAL SERVER ERROR (status code: 500): 500: Internal Server Error
```

**Comment by norrisng** [comment]
This might provide a solution - it attempts to reconnect, with exponential backoff: https://github.com/Rapptz/discord.py/issues/137#issuecomment-198724285

**Comment by norrisng** [comment]
Implemented in d8fdeb67c1e193a05a4c24528c2dd714087afb96.
